admin-manual/maintenance/maintenance.rst

```diff
@@ -134,72 +142,281 @@ The index names are: `aips`, `aipfiles`, `transfers` and `transferfiles`.
+  Storage Service is not running by stopping the ``nginx`` or ``storage-service``
+  services or by making the backup at a time that it is not in use.
+
+To restore Storage Service from backup:
+
+.. code:: bash
+
+  service archivematica-storage-service stop
+  rsync -av /backup/location/storage.db /var/archivematica/storage-service/storage.db
+  rsync -av /backup/location/storage_service /var/archivematica/storage_service
+  service archivematica-storage-service start
+
+Elasticsearch
+^^^^^^^^^^^^^
+
 Elasticsearch is used to store long-term data. Instructions and scripts for
 backing up and restoring Elasticsearch are available in the
 `Elasticsearch documentation`_.
 
+**Preconfiguration**
+
+The path.repo and snapshot repository have to be configured. For example, using
+``/var/lib/elasticsearch/backup-repo`` as the repo path:
+
+.. code:: bash
+
+  mkdir /var/lib/elasticsearch/backup-repo
+  chmod 0755 /var/lib/elasticsearch/backup-repo
+  chown elasticsearch:elasticsearch /var/lib/elasticsearch/backup-repo
+
+Add this line to the ``/etc/elasticsearch/elasticsearch.yml`` file:
+
+.. code:: bash
+
+  path.repo: /var/lib/elasticsearch/backup-repo
+
+Restart elasticsearch: 
+
+.. code:: bash
+
+  service elasticsearch restart
+
+To use a new directory as snapshot repository, create and adjust permissions for one, like so:
+
+.. code:: bash
+
+  mkdir /var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME
+  chmod 0755 /var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME
+  chown elasticsearch:elasticsearch /var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME
+
+Before any snapshot or restore operation can be performed, a snapshot repository
+should be registered in Elasticsearch. The repository settings are
+repository-type specific:
+
+.. code:: bash
+
+  curl -XPUT 'http://localhost:9200/_snapshot/es_backup_YOUR-NAME' -d '{
+      "type": "fs",
+      "settings": {
+          "compress" : true,
+          "location": "/var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME"
+      }
+  }'
+
+**Backing up Elasticsearch indexes**
+
+To make a backup (snapshot) for the ``aips``, ``aipfiles``, ``transfer`` and
+``transferfiles`` indexes, a different name has to be used every time a snapshot
+is taken. For example, using the date inside the filename:
+
+.. code:: bash
+
+  curl -XPUT "http://localhost:9200/_snapshot/es_backup_YOUR-NAME/%3Csnapshot-am-%7Bnow%2Fd%7D%3E?wait_for_completion=true" -d'
+  {
+    "indices": "aips,aipfiles,transfers,transferfiles",
+    "ignore_unavailable": true,
+    "include_global_state": false
+  }'
+
+The snapshot will be saved to the
+``/var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME`` directory. This
+directory can be backed up, for example, using rsync:
+
+.. code:: bash
+
+  rsync -av /var/lib/elasticsearch/backup-repo/es_backup_YOUR-NAME /backup/location/elasticsearch
+
+To list all the snapshots:
+
+.. code:: bash
+
+  curl -XGET "https://localhost:9200/_snapshot/es_backup_YOUR-NAME/_all"
```

Comment:
the protocol should be `http` instead of `https`. And better use "pretty=true" option.

```
curl -XGET "http://localhost:9200/_snapshot/es_backup_YOUR-NAME/_all?pretty=true"
```

without `pretty=true` option:

```
root@localhost:~# curl -XGET "http://localhost:9200/_snapshot/es_backup_YOUR-NAME/_all"
{"snapshots":[{"snapshot":"snapshot-am-2020.04.22","uuid":"afDDEGw3QSCOmaqYxc8tmQ","version_id":6050499,"version":"6.5.4","indices":["aipfiles","aips","transfers","transferfiles"],"include_global_state":false,"state":"SUCCESS","start_time":"2020-04-22T17:14:48.118Z","start_time_in_millis":1587575688118,"end_time":"2020-04-22T17:14:49.082Z","end_time_in_millis":1587575689082,"duration_in_millis":964,"failures":[],"shards":{"total":20,"failed":0,"successful":20}}]}root@localhost:~#
```

with `pretty=true` option:

```
root@localhost:~# curl -XGET "http://localhost:9200/_snapshot/es_backup_YOUR-NAME/_all?pretty=true"
{
  "snapshots" : [
    {
      "snapshot" : "snapshot-am-2020.04.22",
      "uuid" : "afDDEGw3QSCOmaqYxc8tmQ",
      "version_id" : 6050499,
      "version" : "6.5.4",
      "indices" : [
        "aipfiles",
        "aips",
        "transfers",
        "transferfiles"
      ],
      "include_global_state" : false,
      "state" : "SUCCESS",
      "start_time" : "2020-04-22T17:14:48.118Z",
      "start_time_in_millis" : 1587575688118,
      "end_time" : "2020-04-22T17:14:49.082Z",
      "end_time_in_millis" : 1587575689082,
      "duration_in_millis" : 964,
      "failures" : [ ],
      "shards" : {
        "total" : 20,
        "failed" : 0,
        "successful" : 20
      }
    }
  ]
}
```